porch/pkg/git/git.go

```diff
@@ -962,6 +962,35 @@ func (r *gitRepository) loadTasks(ctx context.Context, startCommit *object.Commi
 	return tasks, nil
 }
 
+func (r *gitRepository) getResources(hash plumbing.Hash) (map[string]string, error) {
+	resources := map[string]string{}
+
+	tree, err := r.repo.TreeObject(hash)
+	if err == nil {
```

Comment:
Nit: We should warn if the hash isn't found, and we should also check the error

Reply:
Any reason to treat hash not found different from other errors, other than perhaps trying to get a better message to the user? It seems like almost no matter what goes wrong here, an error would be a better than returning an empty map.